kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/OpenShiftStartupStrategy.java

```diff
@@ -0,0 +1,207 @@
+import org.kie.server.services.impl.KieServerImpl;
+import org.kie.server.services.impl.storage.KieServerState;
+import org.kie.server.services.openshift.impl.storage.cloud.CloudClientFactory;
+import org.kie.server.services.openshift.impl.storage.cloud.KieServerStateCloudRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenShiftStartupStrategy implements StartupStrategy {
+
+    private static final String KIE_SERVER_ROLLOUT_IN_PROGRESS = "kieserver-rollout-in-progress";
+    private static final String KIE_SERVER_INSTANCE_ID = "kieserver-" + UUID.randomUUID().toString();
+    private static final String ROLLOUT_TRIGGER_TIMESTAMP = "services.server.kie.org/openshift-startup-strategy.changeTimestamp";
+    private static final Logger logger = LoggerFactory.getLogger(OpenShiftStartupStrategy.class);
+
+    private static Supplier<OpenShiftClient> clouldClientHelper = () -> (new CloudClientFactory() {}).createOpenShiftClient();
+
+    private static class WatchRunner implements Runnable {
+
+        private boolean isWatchRunning = true;
+        private String kieServerId;
+
+        public WatchRunner(String serverId) {
+            kieServerId = serverId;
+        }
+
+        @Override
+        public void run() {
+            try (OpenShiftClient client = clouldClientHelper.get()) {
+
+                logger.info("Watching ConfigMap in namespace {}", client.getNamespace());
+
+                try (Watch watchable = client.configMaps().watch(new Watcher<ConfigMap>() {
+
+                    @Override
+                    public void eventReceived(Action action, ConfigMap kieServerState) {
+
+                        logger.info("Event - Action: {}, {} on ConfigMap ",
+                                    action, kieServerState.getMetadata().getName());
+
+                        DeploymentConfig dc = client.deploymentConfigs().withName(kieServerId).get();
+
+                        if (kieServerId.equals(kieServerState.getMetadata().getName()) 
+                                && action.equals(Action.MODIFIED) 
+                                && "True".equals(dc.getStatus().getConditions().get(0).getStatus()) 
+                                && triggerRollout(client, kieServerId)) {
+
+                            ObjectMeta md = dc.getSpec().getTemplate().getMetadata();
+                            Map<String, String> ann = md.getAnnotations() == null ? new HashMap<>() : md.getAnnotations();
+                            md.setAnnotations(ann);
+                            ann.put(ROLLOUT_TRIGGER_TIMESTAMP,
+                                    ZonedDateTime.now().format(DateTimeFormatter.ISO_INSTANT));
+                            client.deploymentConfigs().createOrReplace(dc);
+
+                            logger.info("Update DeploymentConfig: {}", md.getName());
+                        } else {
+                            logger.info("Event - Ignored");
```

Comment:
not sure how often this can be logged so maybe change the level to debug and also log the event that is being ignored?

Reply:
The log entry volume for events triggering DC roll-out should NOT be overwhelmingly large. Keeping it at 'Info' level will allow operation to have better monitoring on DC roll-outs. Changing the log level to 'debug' for 'Ignored events'.